homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any
+
+from huum.const import SaunaStatus
+from huum.exceptions import SafetyException
+from huum.huum import Huum
+from huum.schemas import HuumStatusResponse
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huum sauna with config flow."""
+    huum_handler = hass.data.setdefault(DOMAIN, {})[entry.entry_id]
+
+    async_add_entities([HuumDevice(huum_handler, entry.entry_id)], True)
+
+
+class HuumDevice(ClimateEntity):
+    """Representation of a heater."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_max_temp = 110
+    _attr_min_temp = 40
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _target_temperature = 40
+    _status: HuumStatusResponse | None = None
+
+    def __init__(self, huum_handler: Huum, unique_id: str) -> None:
+        """Initialize the heater."""
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name="Huum sauna",
+            manufacturer="Huum",
+        )
+
+        self._huum_handler = huum_handler
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        if self._status and self._status.status == SaunaStatus.ONLINE_HEATING:
+            return HVACMode.HEAT
+        return HVACMode.OFF
+
+    @property
+    def icon(self) -> str:
+        """Return nice icon for heater."""
+        if self.hvac_mode == HVACMode.HEAT:
+            return "mdi:radiator"
+        return "mdi:radiator-off"
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return the current temperature."""
+        return getattr(self._status, "temperature", None)
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return the temperature we try to reach."""
+        return getattr(self._status, "target_temperature", None)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set hvac mode."""
+        if not self.target_temperature:
+            raise HomeAssistantError("No target temperature set, can't turn on sauna")
```

Comment:
Right so let me get this straight. What actions can you execute on the sauna and what will change? If I understand correctly, when the sauna is idle, everything is None. Can you set temperature without setting hvac mode? can you set hvac mode without setting temperature?

I think recently we also introduced the generic `turn_on` and `turn_off` functions for climate. Could they be useful?

Reply:
> Can you set temperature without setting hvac mode?
This is not possible unfortunately

> Can you set hvac mode without setting temperature?
Also no, they are connected, you set the temperature at the same time as turning it on

Reply:
Should they be sent combined to the device? Or can they be separated and only when both are sent it will turn on?

Reply:
While I have not looked in to `turn_on`  and `turn_off` I don't really see them as being helpful here. The mode `HEAT` is kinda what you do and you turn on the sauna, and it makes the UX niced for the user as well when the UI looks like this
<img width="490" alt="image" src="https://github.com/home-assistant/core/assets/3242350/619e04e6-92c1-4c7a-9ba7-9ef86e0c322c">

whereas "on" and "off" would not really signal the same thing. But I might be wrong here and the on/off might do more than I know about.

Reply:
Hmm, so in that light it makes sense to set a temperature directly when you select a HVAC mode.

Reply:
>Should they be sent combined to the device? Or can they be separated and only when both are sent it will turn on?

They are sent combined yes, the "on" command also sets the temperature at the same time. There is no way (at least with the API docs I know of) to set the temperature separately from turning on the sauna. So turning it on and settings the temp is the same action in their API.

Reply:
> Hmm, so in that light it makes sense to set a temperature directly when you select a HVAC mode.

By this, do you want me to refactor things back so that it does try to set the min temperature if no temperature is set, or should we just rely on the status to have been fetched by the time a user tried to turn on the sauna. The user would need to be mighty fast in order to turn on the sauna in between settings up the integration and the first `async_update` call I would say.

Reply:
Honestly, I think you know better what the use case would be. I think its fine to turn it on on the minimum temperature.

When I responded that it should be removed was because I did not know you needed both to actually work so sorry for that.

Reply:
This should now be resolved, I'm using the minimum temp if there is no target temp set. This should only be the case pretty much if someone has never used the sauna ever, and sets it up in home assistant and turns it on there before they try to do so manually. So this should be a uber-corner case if I'm not missing something, but it is now taken into consideration 😄